src/service/action-impl.js

```diff
@@ -766,6 +761,15 @@ export class ActionService {
   }
 }
 
+/**
+ * @param {string} lowercaseTagName
+ * @return {boolean}
+ * @private
+ */
+function isAmpTagName(lowercaseTagName) {
+  return lowercaseTagName.substring(0, 4) === 'amp-';
```

Comment:
[`isAmpElement`](https://github.com/ampproject/amphtml/blob/1672ae0954ce78a2ae3d4d803b7e98943ef17894/src/dom.js#L729-L741)

Reply:
In the second usage we only have a string (attribute value).

Reply:
Extract out of `isAmpElementTagName`? Either way, you're missing `amp-body` and `amp-sticky-ad-top-padding`, which are not AMP elements.

Reply:
Semantics for actions are a bit more complex/confusing e.g. `<script id="amp-access">` is not an AMP element but we allow invoking actions on it, so I'd rather not conflate the two by calling `isAmpElement()` or similar. 

Maybe it should be called "isAmpTarget" rather than "isAmpTagName" though.

Tangent: I actually think `isAmpElement`'s explicit check for `amp-body` etc. isn't great. We should probably duck-type check instead e.g. `!!element.isLayoutSupported`.

Reply:
> Tangent: I actually think isAmpElement's explicit check for amp-body etc. isn't great. We should probably duck-type check instead e.g. !!element.isLayoutSupported.

It's intentional because we can't trust that the element has been upgraded already.